content/6_Advanced/String_Search.mdx

```diff
@@ -175,9 +175,7 @@ We would then have these two cases:
 #include <bits/stdc++.h>
 using namespace std;
 
-int main() {
-	string s;
-	cin >> s;
+vector<int> z_function(const string &s) {
 	vector<int> z(s.size());
 
 	for (int i = 1, l = 0, r = 0; i < s.size(); i++) {
```

Comment:
i think it would look nicer if the `l = 0` and `r = 0` came before the loop, like how it is in the python version

Reply:
![](https://media1.tenor.com/m/6rmxp2-dNrcAAAAC/class-of-09-this-is-true.gif)

Reply:
Fixed!